qcelemental/molparse/to_schema.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+import numpy as np
+
+import qcelemental as qcel
+
+from ..util import unnp
+from ..exceptions import *
+from .to_string import formula_generator
+
+def to_schema(molrec, dtype, units='Angstrom', return_type='json'):
+    """Translate Psi4 json Molecule spec into json from other schemas.
+
+    Parameters
+    ----------
+    molrec : dict
+        Psi4 json Molecule spec.
+    dtype : {'psi4', 1}
+        Molecule schema format.
+    units : {'Angstrom', 'Bohr'}
+        Units in which to write string. There is not an option to write in
+        intrinsic/input units. Some `dtype` may not allow all units.
+    return_type : {'json', 'yaml'}
+        Serialization format string to return.
+
+    Returns
+    -------
+    qcschema : str
+        String of the `return_type` repr of `molrec`.
+
+    """
+    qcschema = {}
+
+    if molrec['units'] == 'Angstrom' and units == 'Angstrom':
+        factor = 1.
+    elif molrec['units'] == 'Angstrom' and units == 'Bohr':
+        if 'input_units_to_au' in molrec:
+            factor = molrec['input_units_to_au']
+        else:
+            factor = 1. / qcel.constants.bohr2angstroms
+    elif molrec['units'] == 'Bohr' and units == 'Angstrom':
+        factor = qcel.constants.bohr2angstroms
+    elif molrec['units'] == 'Bohr' and units == 'Bohr':
+        factor = 1.
+    else:
+        raise ValidationError("""units must be 'Angstrom'/'Bohr', not {}""".format(units))
+    geom = np.array(molrec['geom']) * factor
+    nat = geom.shape[0] // 3
+
+    name = molrec.get('name', formula_generator(molrec['elem']))
+#    tagline = """auto-generated by qcdb from molecule {}""".format(name)
+
+    if dtype == 'psi4':
+        qcschema = copy.deepcopy(molrec)
+        qcschema['geom'] = geom
+        qcschema['units'] = units
+        qcschema['name'] = name
+
+    elif dtype == 1:
```

Comment:
What is a 1 dtype?

Reply:
You wanted simple versions for QCSchema, so this is it at version `1`. `dtype='psi4'` is the expanded dict that (a) has a much larger required fields set, (b) has slightly different systematic field names (elez, elea, elem, etc.), and (c) is the intermediate repr that molparse works with (result of `Molecule.to_dict()`).

And, yeah, QCSchema gets [shut down fast](https://github.com/qcdb/QCElemental/blob/molparse/qcelemental/molparse/to_schema.py#L59-L61) if you try to ask for it in non-Bohr

Reply:
Oh gotcha, dtype is schema version.